### DISCUSSION TITLE
Open AI Access Key Issue

### DISCUSSION THREAD

**Discussion by pruthvirajv** [microsoft/AI-Chat-App-Hack#145] [Q&A]
Dear Team,

When I try to debug the code on local VS Code, I am getting the following error. Could you please let me know if there is any fix

openai.AuthenticationError: Error code: 401 - {'error': {'code': 'PermissionDenied', 'message': 'Principal does not have access to API/Operation.'}}

**Comment by pamelafox** [comment]
That typically means that your user roles did not get set up properly. This is after you run "azd up"? Are you still logged in as the same user as when you ran "azd up"?

**Reply by pruthvirajv** [reply]
Hi Pamela,

Many thanks for your response. Yes, this happened after running the azd up. I am logged in as the same user who created the resources. 

Could you please let me know the user roles that I need to check accessing the frond end hosted in the Azure web app is perfectly working fine whereas if I try to debug from VS code this issue is shown.

Regards,
Pruthvi

**Reply by pamelafox** [reply]
Your user account should have the roles created here:
https://github.com/Azure-Samples/azure-search-openai-demo/blob/9e76dc3618347afb937dc6b9496217d20ae683e0/infra/main.bicep#L459

Specifically, this one:

module openAiRoleUser 'core/security/role.bicep' = if (isAzureOpenAiHost) {
  scope: openAiResourceGroup
  name: 'openai-role-user'
  params: {
    principalId: principalId
    roleDefinitionId: '5e0bd9bd-7b93-4f28-af87-19fc36ad61bd'
    principalType: principalType
  }
}

You can check that in the Portal with the access control tab:

![Screenshot 2024-04-03 at 12 51 37 PM](https://github.com/microsoft/AI-Chat-App-Hack/assets/297042/380ec8d1-6e1f-4d66-b245-c223ec0b613b)
